db refine

fluiderson opened the pull request.

fluiderson committed:
db format done

fluiderson committed:
connect new db module with everything

fluiderson committed:
refact, api spec, cb, withdrawal

fluiderson committed:
rm debug msg

Vovke commented:
by doing this "folder(chain2) based code backup strategy" you will eventually override the commit history of the code. Please ensure that it's not happening.

fluiderson committed:
fix storage fetcher, fix cs configs & add zym.toml

fluiderson committed:
remove token name checks, support both native & asset tokens, ignore the empty response from balance requests

fluiderson committed:
fix invoice death

Slesarew commented on src/signer.rs:
```diff
@@ -1,157 +1,249 @@
+    }
+
+    pub fn sign(&self, id: impl AsRef<[u8]>, msg: &[u8]) -> Result<Signature, SignerError> {
+        let mut random = rand::thread_rng();
```
seriously? This is exactly why you should not be implementing any of this manually. This is supposed to internally call OsRnd which, by the way, IS blocking, thus threading here.

fluiderson replied:
Do you mean that `rand::thread_rng()` does block for too long, or?.. If you do, then I don't quite understand for what we need to save so much time. And by the way, we already use the same function [here](https://github.com/paritytech/soketto/blob/cd261c9bf22e4177cc70deddd1c8aabc40715a51/src/handshake/client.rs#L133) on every payout (as we create a new connection each time).

Slesarew committed:
Merge pull request #45 from Slesarew/new-api-mend

major: api_v2 and major rehaul of everything

Vovke committed:
feat: updating create order to match with API specs (with major refactoring around)

create order with json payload

Refactoring main.rs moving to lib.rs before tests part 1/2

cleaned up some warnings

Refactoring main.rs moving to lib.rs before tests part 2/2

chore: bump version

Vovke committed:
chore: bump version

Vovke committed:
fix: native token broken parsing

Slesarew committed:
Merge pull request #54 from Vovke/api-specs

feat: updating create order to match with API specs (with major refactoring around)

Vovke committed:
green fmt check

Slesarew committed:
Merge pull request #65 from Vovke/fmt

chore: fmt

dependabot[bot] committed:
build(deps): bump actions/checkout from 4.1.2 to 4.2.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.2 to 4.2.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4.1.2...v4.2.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Slesarew committed:
Merge pull request #66 from Alzymologist/dependabot/github_actions/actions/checkout-4.2.0

build(deps): bump actions/checkout from 4.1.2 to 4.2.0

fluiderson committed:
commit

fluiderson committed:
Merge branch 'main' into db-refine

fluiderson committed:
finish server & db definitions, add chain hashes to connected chains

fluiderson commented:
Moved to https://github.com/Alzymologist/Kalatori-backend/pull/71.
